src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_DocumentHighlights.cs

```diff
@@ -17,20 +17,23 @@ internal partial class CodeAnalysisService : IRemoteDocumentHighlights
         public async Task<IList<SerializableDocumentHighlights>> GetDocumentHighlightsAsync(
             DocumentId documentId, int position, DocumentId[] documentIdsToSearch, CancellationToken cancellationToken)
         {
-            // NOTE: In projection scenarios, we might get a set of documents to search
-            // that are not all the same language and might not exist in the OOP process
-            // (like the JS parts of a .cshtml file). Filter them out here.  This will
-            // need to be revisited if we someday support FAR between these languages.
-            var solution = await GetSolutionAsync(cancellationToken).ConfigureAwait(false);
-            var document = solution.GetDocument(documentId);
-            var documentsToSearch = ImmutableHashSet.CreateRange(
-                documentIdsToSearch.Select(solution.GetDocument).WhereNotNull());
+            return await RunServiceAsync(async () =>
```

Comment:
why are some RunServiceAsync calls awaited, but some are not?

Reply:
Task doesn't provide co/contra-variant. when return type is different, I can either cast or add await. I added await.